ompi/mca/io/ompio/io_ompio_file_write.c

```diff
@@ -371,12 +371,21 @@ int mca_io_ompio_file_write_all_begin (ompi_file_t *fh,
     data = (mca_common_ompio_data_t *) fh->f_io_selected_data;
     fp = &data->ompio_fh;
     if ( true == fp->f_split_coll_in_use ) {
-	printf("Only one split collective I/O operation allowed per file handle at any given point in time!\n");
-	return MPI_ERR_OTHER;
+        printf("Only one split collective I/O operation allowed per file handle at any given point in time!\n");
+        return MPI_ERR_OTHER;
+    }
+
+    OPAL_THREAD_LOCK(&fh->f_lock);
+    if (true == fp->f_split_coll_in_use) {
+        printf("Only one split collective I/O operation allowed per file handle at any given point in time!\n");
+        OPAL_THREAD_UNLOCK(&fh->f_lock);
+        return MPI_ERR_REQUEST;
```

Comment:
This check seems to be duplicated from above?

Reply:
no, its not really a duplicate, the first one is before you acquire the lock (to avoid racing for the lock itself if possible), the second one is after you the lock has been acquired to catch a potential race on that level, since these are non-blocking operation. The same sequence  (check before lock, lock, check after lock) is used in multiple places in ompi